pkg/pipelines/tekton/gitlab_test.go

```diff
@@ -0,0 +1,579 @@
+//go:build integration
+// +build integration
+
+package tekton_test
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/xanzy/go-gitlab"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/net/html"
+
+	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"knative.dev/pkg/apis"
+
+	"knative.dev/func/pkg/docker"
+	fn "knative.dev/func/pkg/functions"
+	"knative.dev/func/pkg/k8s"
+	"knative.dev/func/pkg/pipelines"
+	"knative.dev/func/pkg/pipelines/tekton"
+	"knative.dev/func/pkg/random"
+)
+
+const (
+	gitlabURL        = "http://gitlab.127.0.0.1.sslip.io"
+	pacControllerURL = "http://pac-ctr.127.0.0.1.sslip.io"
```

Comment:
+1

Reply:
@lkingland how would we ensure proper resolution without messing with /etc/hosts on developer machine?

Reply:
Beauty of `*.127.0.0.1.sslip.io` is that it resolves to localhost on developer machine without modifying `/etc/hosts/`.

Reply:
Also we have been using `127.0.0.1.sslip.io` for knative service routes for long time in tests.

Reply:
>Can we please update this to be gitlab.cluster.local and pac-ctr.cluster.local?

I would prefer not to.

>This could even be as simple as passing in a GITLAB_ADDRESS PAC_CONTROLLER_ADDRESS for now.

This should be doable.